### PR TITLE
Fix version_table to work without Cython installed

### LINKED ISSUE #2001: Clean up qutip.ipynbtools for version 5
### Bug Description

In https://github.com/qutip/qutip-tutorials/pull/42#pullrequestreview-1119409348 @BoxiLi noticed that `qutip.ipynbtools` imports Cython without a `try: ... except: ...` around the import. This results in `qutip.ipynbtools.version_table()` failing even though Cython isn't a required dependency of QuTiP.

We should do a thorough clean up of this module in dev-major and fix this particular issue while we're there.

### Code to Reproduce the Bug

_No response_

### Code Output

_No response_

### Expected Behaviour

`qutip.ipynbtools.version_table()` can be called without Cython installed.

### Your Environment

```shell
QuTiP version: 5.0.0.dev
```

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by gadhvirushiraj**
fix #2001

**Description**
- added `try: ...except:` in import of `Cython`
- fixed `version_table` to work without Cython installed

**Behaviour**
`qutip.ipynbtools.version_table()` can be called without Cython installed

### CODE REVIEW

File: qutip/ipynbtools.py
Comment (AGaliciaMartinez) [inline qutip/ipynbtools.py:48]:
```suggestion
except ImportError:
    pass
```

For try except it is best to be specific with the type of exception to avoid unexpected bugs.

**Review by AGaliciaMartinez — CHANGES_REQUESTED** [review]
Hi @gadhvirushiraj,

Thank you for the fix! We will need some tests to make sure that things work as intended. Our CI tools test several environments, including some with and without Cython. We would need to write a test for this, which is a little bit tricky in this case but Im sure you can handle it :smile:. This would be located in a new file in `qutip/tests/test_ipynbtools.py`. You will need to call the function `version_table` and make sure, for example, that Cython only appears in the output data if `Cython` is in `sys.modules`.  For this you will need to inspect the output string which is contained under the `html.data` attribute of the `HTML` object.

Feel free to ask for help if you get stuck!

**Comment by gadhvirushiraj** [comment]
Hello @AGaliciaMartinez,
Thanks for the review. I will look into it and update.

File: qutip/ipynbtools.py
Comment (hodgestar) [inline qutip/ipynbtools.py:49]:
```suggestion
try:
    import Cython
    _cython_available = True
except ImportError:
    _cython_available = False

```
Let's rather explicitly record whether the import of Cython succeeded. This is much more obviously correct than digging in `sys.modules` where all kinds of strange things may occur.

File: qutip/ipynbtools.py
Comment (hodgestar) [inline qutip/ipynbtools.py:82]:
```suggestion
    if _cython_available:
        packages.append(("Cython", Cython.__version__))
```

File: qutip/ipynbtools.py
Comment (hodgestar) [inline qutip/ipynbtools.py:377]: Please put back the new line at the end of the file.

File: qutip/tests/test_ipynbtools.py
Comment (hodgestar) [inline qutip/tests/test_ipynbtools.py:5]: Let's split this into two tests:

- `test_version_table` with is always called and sanity checks the version table.
- `test_version_table_with_cython` which is only called when Cython is installed.

E.g. 
```
@pytest.mark.parametrize('verbose', [False, True])
def test_version_table(verbose):
    html = version_table(verbose=verbose)
    assert  "<th>Software</th>" in html.data
    assert "<th>Version</th>" in html.data
    if verbose:
        assert "<td>Installation path</td>" in html.data


@pytest.importorskip("Cython")
def test_version_table_with_cython():
    html = version_table()
    assert "<td>Cython</td>" in html.data
```

**Review by hodgestar — CHANGES_REQUESTED** [review]
Thank you for your PR. I've left some suggestions for improving the implementation and tests. Once you've implemented them, please ping us for a second review.

File: qutip/tests/test_ipynbtools.py
Comment (gadhvirushiraj) [reply qutip/tests/test_ipynbtools.py:5]: Hello @hodgestar,
Thanks for the review. 
Should I also check for other library dependencies under the `test_version_table` ?

Comment (hodgestar) [reply qutip/tests/test_ipynbtools.py:5]: Yes, adding a few more checks to `test_version_table` would be good.

File: qutip/ipynbtools.py
Comment (hodgestar) [inline qutip/ipynbtools.py:81]:
```suggestion

```

**Review by hodgestar — APPROVED** [review]

**Comment by hodgestar** [comment]
@gadhvirushiraj Thanks! I've fixed the Code Climate issue and approved. I'll merge once the CI checks finish (and pass, which they should do).

**Pull request merged**
